### PR TITLE
Default to development in dev

### PR DESCRIPTION
**Pull request by benbalter**
Fixes https://github.com/jekyll/github-metadata/issues/43.

Also fixes the domain to use the Pages hostname, not the GitHub hostname, when custom domains are disabled in order to get tests to pass.

### CODE REVIEW

**Comment by parkr** [comment]
@benbalter Isn't that supposed to be `JEKYLL_ENV`? `PAGES_ENV` is used to determine the target to request from, I think.

**Comment by benbalter** [comment]
> PAGES_ENV is used to determine the target to request from, I think.

That doesn't make much sense to me. The idea behind `PAGES_ENV` was (A) so that we could know if we were in enterprise or dotcom internally, and (B) so that the user could know if they were in dotcom (or enterprise) or locally. I believe it predates `JEKYLL_ENV`. Additionally, is `JEKYLL_ENV` exposed to liquid where the user can access it? 

If we thought of using `PAGES_ENV` as a flag/toggle, it'd really only be useful, if I could set `PAGES_ENV` to the enterprise instance, and everything else just worked. Otherwise, I can't see when a user would want to actively toggle back and forth. Even if pushing to both remotes, they'd want to render things differently locally (e.g., changing the URL, not including discuss, etc.).

In my mind:

* `PAGES_ENV`: should be `dotcom`, `enterprise`, or `development` (default)
* `JEKYLL_ENV: should be `production` or `development`

The two are similar, but `PAGES_ENV` is exposed to liquid, and can be used to differentiate between GitHub environments.

**Comment by parkr** [comment]
@benbalter `jekyll.environment` exposes `JEKYLL_ENV`. It too defaults to `development`. The only difference I see is that `PAGES_ENV` differentiates between `dotcom` and `enterprise`, which are both `production` for `JEKYLL_ENV`.

What should `PAGES_ENV=development` do about changing the octokit API calls? `https://github.dev`?

**Comment by benbalter** [comment]
> What should PAGES_ENV=development do about changing the octokit API calls? https://github.dev?

I think nothing? As someone previewing my Pages-hosted site locally, I'd want the metadata (save the URL) to be the live metadata, no?

**Comment by parkr** [comment]
> I think nothing? As someone previewing my Pages-hosted site locally, I'd want the metadata (save the URL) to be the live metadata, no?

What if it's an Enterprise repo? Wouldn't that affect this? Would we have `development` and `development-enterprise`?

**Comment by benbalter** [comment]
> What if it's an Enterprise repo?

That's what I was asking about in https://github.com/jekyll/github-metadata/issues/48. (and suggested ENDPOINT)

File: lib/jekyll-github-metadata/repository.rb [hunk 1]
```diff
@@ -172,7 +172,7 @@ def cname
       def domain
         @domain ||=
           if !Pages.custom_domains_enabled?
-            Pages.github_hostname
+            Pages.pages_hostname
```
Comment (parkr) [inline lib/jekyll-github-metadata/repository.rb]: An associated test may need to be updated for this.

**Comment by benbalter** [comment]
@parkr got tests passing, if you're able to :eyes: once more. :smile: Would be good to finally get this out.

**Comment by parkr** [comment]
LGTM.

**Comment by benbalter** [comment]
LGTM.

**Comment by benbalter** [comment]
@jekyllbot: merge +major

**Pull request merged**